### PR TITLE
[CALCITE-5723] Oracle dialect generates SQL that cannot be recognized by lower version Oracle Server(<12) when unparsing OffsetFetch

### PR DESCRIPTION
**Pull request by ILuffZhe**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -7089,7 +7105,18 @@ Sql withMysql8() {
...
+      return withOracle(12);
+    }
+
+    Sql withOracle(int majorVersion) {
+      final SqlDialect oracleDialect = DatabaseProduct.ORACLE.getDialect();
+      return dialect(
+          new OracleSqlDialect(OracleSqlDialect.DEFAULT_CONTEXT
+              .withDatabaseProduct(DatabaseProduct.ORACLE)
+              .withDatabaseMajorVersion(majorVersion)
+              .withIdentifierQuoteString(oracleDialect.quoteIdentifier("")
```
Comment (JiajunBernoulli) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:7117]: Why need `withIdentifierQuoteString` and `withNullCollation`?

Comment (ILuffZhe) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:7117]: In this case, we don't need those two configuration. But I'm wandering if they might be reused in the future cases? And I also see that both `withMssql` and `withMysql8` do the same thing.

File: core/src/main/java/org/apache/calcite/sql/dialect/OracleSqlDialect.java [hunk 2]
```diff
@@ -188,4 +191,13 @@ public OracleSqlDialect(Context context) {
...
     }
   }
+
+  @Override public void unparseOffsetFetch(SqlWriter writer, @Nullable SqlNode offset,
+       @Nullable SqlNode fetch) {
+    // majorVersion in SqlDialect.EMPTY_CONTEXT is -1 by default
+    if (this.majorVersion != -1 && this.majorVersion < 12) {
+      throw new RuntimeException("Lower Oracle version(<12) doesn't support offset/fetch syntax!");
+    }
+    unparseFetchUsingAnsi(writer, offset, fetch);
```
Comment (JiajunBernoulli) [inline core/src/main/java/org/apache/calcite/sql/dialect/OracleSqlDialect.java:201]: `super.unparseOffsetFetch(writer, offset, fetch))` is better for me.

Comment (ILuffZhe) [reply core/src/main/java/org/apache/calcite/sql/dialect/OracleSqlDialect.java]: Make sense, I've updated the PR.

**Review by JiajunBernoulli — APPROVED** [review]

**Comment by ILuffZhe** [comment]
@JiajunBernoulli  Thanks for your review!

**Comment by ILuffZhe** [comment]
Squash the commits into one.

**Pull request merged**